File: backend/src/main/java/com/minor/alumini_platform/model/Alumni.java
package com.minor.alumini_platform.model;

import java.util.List;

import com.fasterxml.jackson.annotation.JsonBackReference;
import com.fasterxml.jackson.annotation.JsonManagedReference;
import com.minor.alumini_platform.enums.EmploymentStatus;

import jakarta.persistence.*;

@Entity
@Table(name = "alumni")
public class Alumni {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false, unique = true)
    private String enrollmentNumber;
    @Column(nullable = false)
    private String name;
    @Column(nullable = false)
    private String email;
    @Column(nullable = false)
    private String password;
    @Column(nullable = false)
    private String passingYear;
    @Column(nullable = false)
    private String department;

    @Enumerated(EnumType.STRING)
    private EmploymentStatus employmentStatus;

    @OneToMany(mappedBy = "alumni", cascade = CascadeType.ALL, orphanRemoval = true, fetch = FetchType.EAGER)
    @JsonManagedReference
    private List<Experience> experiences;

    @OneToMany(mappedBy = "alumni", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<Post> posts;

    // Getters & Setters
    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getEnrollmentNumber() {
        return enrollmentNumber;
    }

    public void setEnrollmentNumber(String enrollmentNumber) {
        this.enrollmentNumber = enrollmentNumber;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getEmail() {
        return email;
    }

    public void setEmail(String email) {
        this.email = email;
    }

    public String getPassword() {
        return password;
    }

    public void setPassword(String password) {
        this.password = password;
    }

    public String getPassingYear() {
        return passingYear;
    }

    public void setPassingYear(String passingYear) {
        this.passingYear = passingYear;
    }

    public String getDepartment() {
        return department;
    }

    public void setDepartment(String department) {
        this.department = department;
    }

    public EmploymentStatus getEmploymentStatus() {
        return employmentStatus;
    }

    public void setEmploymentStatus(EmploymentStatus employmentStatus) {
        this.employmentStatus = employmentStatus;
    }

    public List<Experience> getExperiences() {
        return experiences;
    }

    public void setExperiences(List<Experience> experiences) {
        this.experiences = experiences;
    }

    public List<Post> getPosts() {
        return posts;
    }

    public void setPosts(List<Post> posts) {
        this.posts = posts;
    }

}

File: frontend/lib/models/alumni.dart
class Alumni {
  final int id;
  final String enrollmentNumber;
  final String name;
  final String email;
  final String password;
  final String passingYear;
  final String department;
  final String? employmentStatus;
  final List<Experience>? experiences;

  Alumni({
    required this.id,
    required this.enrollmentNumber,
    required this.name,
    required this.email,
    required this.password,
    required this.passingYear,
    required this.department,
    this.employmentStatus,
    this.experiences,
  });

  String get fullName => name;

  factory Alumni.fromJson(Map<String, dynamic> json) {
    return Alumni(
      id: json['id'] ?? 0,
      enrollmentNumber: json['enrollmentNumber'] ?? '',
      name: json['name'] ?? '',
      email: json['email'] ?? '',
      password: json['password'] ?? '',
      passingYear: json['passingYear'] ?? '',
      department: json['department'] ?? '',
      employmentStatus: json['employmentStatus'],
      experiences: json['experiences'] != null
          ? (json['experiences'] as List)
              .map((e) => Experience.fromJson(e))
              .toList()
          : null,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'enrollmentNumber': enrollmentNumber,
      'name': name,
      'email': email,
      'password': password,
      'passingYear': passingYear,
      'department': department,
      'employmentStatus': employmentStatus,
      'experiences': experiences?.map((e) => e.toJson()).toList(),
    };
  }
}

class Experience {
  final int? id;
  final String companyName;
  final String position;
  final String startDate;
  final String? endDate;
  final String? description;

  Experience({
    this.id,
    required this.companyName,
    required this.position,
    required this.startDate,
    this.endDate,
    this.description,
  });

  factory Experience.fromJson(Map<String, dynamic> json) {
    return Experience(
      id: json['id'],
      companyName: json['companyName'] ?? '',
      position: json['position'] ?? '',
      startDate: json['startDate'] ?? '',
      endDate: json['endDate'],
      description: json['description'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'companyName': companyName,
      'position': position,
      'startDate': startDate,
      'endDate': endDate,
      'description': description,
    };
  }
}
